src/plugins/data/public/ui/dataset_selector/index.tsx

```diff
@@ -18,9 +18,21 @@ const ConnectedDatasetSelector = ({ onSubmit }: ConnectedDatasetSelectorProps) =
   const { services } = useOpenSearchDashboards<IDataPluginServices>();
   const queryString = services.data.query.queryString;
   const [selectedDataset, setSelectedDataset] = useState<Dataset | undefined>(
-    () => queryString.getQuery().dataset || queryString.getDefaultQuery().dataset
+    () => queryString.getQuery().dataset
   );
 
+  useEffect(() => {
+    const subscription = queryString.getUpdates$(0).subscribe((query: Query) => {
```

Comment:
oh is this because we didnt get the first update on page load?

Reply:
Yeah. Since the earlier implementation was skipping the first update, when we are trying to load a saved search , which loads all the things in first update, Those values were not being reflected in Dataset selector and language selector